src/core/dev_api/dimension_tracker.hpp

```diff
@@ -9,12 +9,28 @@
 #include <unordered_set>
 
 #include "openvino/core/dimension.hpp"
+#include "openvino/core/type/element_type.hpp"
 
 namespace ov {
 
 /// \brief Special label value indicate no label set.
 constexpr size_t no_label = 0;
 
+namespace element {
+/// \brief Get element::Type form label type.
+/// \return element::Type of same size in bytes as label_type (size_t)
+inline Type from_label_type() {
```

Comment:
Could you try making this function constexpr? Its logic is based entirely on the information available during the compile time so perhaps you can mark it explicitly constexpr and the compiler might inline it even more.

Reply:
To make it `constexpr`, I need to make it as template set for each `sizeof(size_t)` as `constexpr` cannot have multiple returns.
So probably then more proper solution would be introduce template specializations for element::from<size_t> for mac OS, as for others used on CI the `size_t` is alias to some unsigned integer which is supported by element::from<>.

I assume this as temporary solution and in PR #15010 there alias for `label_t` as uint32_t (or any other supported by tensor type) then this function is not required at all and can be removed.

Depends which PR will be merged first this or #15010, I will update 2nd one accordingly.